src/test/fuzz/coinscache_sim.cpp

```diff
@@ -148,14 +146,13 @@ class CoinsViewBottom final : public CCoinsView
 public:
     std::optional<Coin> GetCoin(const COutPoint& outpoint) const final
     {
-        // TODO GetCoin shouldn't return spent coins
-        if (auto it = m_data.find(outpoint); it != m_data.end()) return it->second;
+        if (auto it = m_data.find(outpoint); it != m_data.end() && !it->second.IsSpent()) return it->second;
```

Comment:
nit: style of this `GetCoin` is still a tiny bit different from the other one (i.e. brace init)